### PR TITLE
feat: display scheduled (pending) rewards

### PR DESCRIPTION
**Pull request by bajtos**
- Integrate with MERidian smart contract to fetch the amount of pending (scheduled) rewards stored in the contract state.

- Periodically call RPC API to update this value.

- Propagate the value to the Station GUI.

Screenshot 2023-11-07 at 16 06 31

### CODE REVIEW

File: renderer/src/pages/Dashboard.tsx [hunk 1]
```diff
@@ -39,10 +39,9 @@ const Dashboard = (): JSX.Element => {
               </p>
             </div>
             <div className="mb-6">
-              <p className="w-fit text-body-3xs text-grayscale-700 uppercase">Total Earnings (coming soon)</p>
-              <p className="w-fit text-header-m font-bold font-number total-earnings" title="total earnings">
-                {totalEarnings > 0 ? totalEarnings : '--'}
-                {totalEarnings > 0 ? <span className="text-header-3xs">FIL</span> : ''}
+              <p className="w-fit text-body-3xs text-grayscale-700 uppercase">Scheduled rewards</p>
+              <p className="w-fit text-header-m font-bold font-number total-earnings" title="scheduled rewards">
+                {scheduledRewards}&nbsp;<span className="text-header-3xs">FIL</span>
```
Comment (bajtos) [inline renderer/src/pages/Dashboard.tsx:44]: Let's find a better description for this UI field.

We want to convey the information that the user is eligible to receive this amount of FIL in the next reward payout.

How about "pending rewards"?

I think we should also add a tooltip to explain what this means. Proposed text:

> The amount you accrued for running your Station. These funds will be sent to your Station Wallet address at the end of the payout epoch (approximately once a month).

File: main/wallet.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ backend.transactions = loadStoredEntries()
 /** @type {Context | null} */
 let ctx = null
 let balance = loadBalance()
+const scheduledRewards = loadScheduledRewards()
```
Comment (juliangruber) [inline main/wallet.js:33]: since this is a const, doesn't that mean `getScheduledRewards` will always return the same number, independent of `updateScheduledRewards`?

File: renderer/src/pages/Dashboard.tsx (hunk 1 above)
Comment (juliangruber) [reply renderer/src/pages/Dashboard.tsx:44]: Do we have a tooltip component ready?

File: main/wallet-backend.js [hunk 3]
```diff
@@ -57,7 +57,9 @@ class WalletBackend {
       ({ seed, isNew } = await this.getSeedPhrase())
     }
     this.provider = new ethers.providers.JsonRpcProvider(
-      'https://api.node.glif.io/rpc/v0'
+      // REVERT ME BEFORE LANDING!
+      'https://api.calibration.node.glif.io/rpc/v0'
+      // 'https://api.node.glif.io/rpc/v0'
```
Comment (bajtos) [inline main/wallet-backend.js:62]: Reminder for myself: revert this change before landing the PR.

File: main/wallet.js (hunk 2 above)
Comment (bajtos) [reply main/wallet.js:33]: Great catch! I used `let` but VS Code auto-fixed this to `const`. I need to double-check why we are not updating this variable.

Comment (bajtos) [reply main/wallet.js]: Fixed in cae4a42

File: renderer/src/pages/Dashboard.tsx (hunk 1 above)
Comment (bajtos) [reply renderer/src/pages/Dashboard.tsx:44]: > > Do we have a tooltip component ready?
> Great question! Isn't there a standard HTML element to "display tooltip on hover"?

LOL :) It's enough to set the "title" attribute.

Screenshot 2023-11-07 at 16 28 10

Of course, it would be nicer to show an icon like (?) that will make this tooltip easier to discover, but IMO that's "nice to have, no time to implement" right now.

WDYT?

Comment (patrickwoodhead) [reply renderer/src/pages/Dashboard.tsx:44]: Let's add it as a ticket but no need to implement before launch.

Comment (patrickwoodhead) [reply renderer/src/pages/Dashboard.tsx:44]: > Let's find a better description for this UI field.
> 
> We want to convey the information that the user is eligible to receive this amount of FIL in the next reward payout.
> 
> How about "pending rewards"?
> 
> I think we should also add a tooltip to explain what this means. Proposed text:
> 
> > The amount you accrued for running your Station. These funds will be sent to your Station Wallet address at the end of the payout epoch (approximately once a month).

"This is the reward total you have accrued since your last payout. Scheduled earnings will be sent to your Station Wallet approximately once a month, provided you have earned more than the payout threshold"

**Review by juliangruber — APPROVED** [review]

**Comment by juliangruber** [comment]
@bajtos can we release this first thing in the morning please?

File: main/wallet-backend.js [hunk 4]
```diff
@@ -286,6 +292,29 @@ class WalletBackend {
...
+    try {
+      return await this.meridian.rewardsScheduledFor(this.address)
+    } catch (/** @type {any} */ err) {
+      if (
+        err?.error?.error instanceof Error &&
+        err.error.error.message.includes(
+          `resolve address ${this.addressDelegated}: actor not found`
+        )
+      ) {
+        // no-op, our address was not assigned any rewards yet
```
Comment (bajtos) [inline main/wallet-backend.js:311]: See the discussion in Slack: 
https://filecoinproject.slack.com/archives/C03PVLW721L/p1699433434940039

When calling `rewardsScheduledFor(address)`, I am getting the following error, where the delegated address `f410[redacted]` is my Station Wallet.

```
CallWithGas failed: 
 call raw get actor: 
  resolution lookup failed (f410[redacted]): 
   resolve address f410[redacted]: 
    actor not found
```

Comment (bajtos) [reply main/wallet-backend.js]: Fixed in [ea21ce3](https://github.com/filecoin-station/desktop/pull/1082/commits/ea21ce3c980fc64bb3dde90903d0a6f0c06a19df)

**Review by juliangruber — APPROVED** [review]
ship it :)

File: renderer/src/pages/Dashboard.tsx (hunk 1 above)
Comment (bajtos) [reply renderer/src/pages/Dashboard.tsx]: > Let's add the question mark as a ticket but no need to implement before launch.

Done: https://github.com/filecoin-station/desktop/issues/1087
